x-pack/platform/plugins/shared/alerting/server/alerts_service/lib/clear_alert_flapping_history.ts

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import type { Logger } from '@kbn/logging';
+import type { QueryDslQueryContainer } from '@elastic/elasticsearch/lib/api/types';
+import {
+  ALERT_RULE_UUID,
+  ALERT_STATUS,
+  ALERT_STATUS_ACTIVE,
+  ALERT_STATUS_RECOVERED,
+  ALERT_UUID,
+  ALERT_FLAPPING,
+  ALERT_FLAPPING_HISTORY,
+  ALERT_RULE_CONSUMER,
+} from '@kbn/rule-data-utils';
+
+type AlertsResult = Array<{ [ALERT_RULE_UUID]: string; [ALERT_UUID]: string }>;
+
+export interface ClearAlertFlappingHistoryParams {
+  indices: string[];
+  ruleIds: string[];
+}
+
+interface ClearAlertFlappingHistoryParamsWithDeps extends ClearAlertFlappingHistoryParams {
+  logger: Logger;
+  esClient: ElasticsearchClient;
+}
+
+const clearAlertFlappingHistoryQuery = (ruleIds: string[]): QueryDslQueryContainer => {
+  const shouldStatusTerms = [ALERT_STATUS_ACTIVE, ALERT_STATUS_RECOVERED].map((status) => {
```

Comment:
we should not need to update recovered alerts, just active alerts where flapping is true